src/Service.Tests/SqlTests/GraphQLMutationTests/GraphQLMutationTestBase.cs

```diff
@@ -40,6 +40,39 @@ public async Task InsertMutation(string dbQuery)
             SqlTestHelper.PerformTestEqualJsonStrings(expected, actual.ToString());
         }
 
+        /// <summary>
+        /// <code>Do: </code> Attempt to insert a new publisher with name not allowed by database policy.
+        /// <code>Check: </code> Mutation fails with expected exception.
+        /// </summary>
+        /// <param name="dbQuery">SELECT query to validate expected result.</param>
+        /// <param name="errorMessage">Expected error message.</param>
+        /// <param name="roleName">Custom client role in whose context this authenticated request will be executed</param>
+        /// <returns></returns>
+        public async Task InsertMutationFailingDatabasePolicy(string dbQuery, string errorMessage, string roleName)
+        {
+            string graphQLMutationName = "createPublisher";
+            string graphQLMutation = @"
+                mutation {
+                    createPublisher(item: { name: ""New publisher"" }) {
+                        id
+                        name
+                    }
+                }
+            ";
+
+            JsonElement result = await ExecuteGraphQLRequestAsync(graphQLMutation, graphQLMutationName, clientRoleHeader: roleName, isAuthenticated: true);
+
+            SqlTestHelper.TestForErrorInGraphQLResponse(
+                result.ToString(),
+                message: errorMessage,
+                statusCode: $"{DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed}"
+            );
+
+            string dbResponse = await GetDatabaseResultAsync(dbQuery);
+            using JsonDocument dbResponseJson = JsonDocument.Parse(dbResponse);
+            Assert.AreEqual(dbResponseJson.RootElement.GetProperty("count").GetInt64(), 0);
```

Comment:
Add a comment describing why you test the dbQuery provided in test parameters: this confirms a db record was not added as a result of the create mutation. Either added here or in the param comments.